src/modules/users/components/TokenActivation/TokenActivationContent/ClaimsTab.tsx

```diff
@@ -1,19 +1,91 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { FormattedMessage, defineMessages } from 'react-intl';
+import { Extension } from '@colony/colony-js';
+
+import { SpinnerLoader } from '~core/Preloaders';
+import { Address } from '~types/index';
+import {
+  useColonyExtensionsQuery,
+  useSubgraphUserMotionTokenEventsQuery,
+} from '~data/generated';
 
 import styles from './TokenActivationContent.css';
 
 const MSG = defineMessages({
   noClaims: {
     id: 'users.TokenActivation.TokenActivationContent.ClaimsTab.noClaims',
-    defaultMessage: 'There are currently no claims',
+    defaultMessage: 'There are no stakes to claim.',
   },
 });
 
-const ClaimsTab = () => {
+export interface ClaimsTabProps {
+  colonyAddress: Address;
+  walletAddress: Address;
+}
+
+const ClaimsTab = ({ colonyAddress, walletAddress }: ClaimsTabProps) => {
+  const { data: extensions } = useColonyExtensionsQuery({
+    variables: { address: colonyAddress },
+  });
+
+  const { installedExtensions } = extensions?.processedColony || {};
+
+  const votingReputationExtension = installedExtensions?.find(
+    ({ extensionId }) => extensionId === Extension.VotingReputation,
+  );
+
+  const {
+    data: userStakedMotions,
+    loading,
+  } = useSubgraphUserMotionTokenEventsQuery({
+    variables: {
+      colonyAddress: colonyAddress?.toLowerCase(),
+      walletAddress: walletAddress?.toLowerCase(),
+      extensionAddress: votingReputationExtension?.address?.toLowerCase() || '',
+      sortDirection: 'desc',
+    },
+  });
+
+  const unclaimedMotions = useMemo(() => {
```

Comment:
I think that using existing query and filtering in these events is good. It doesn't seem that we really have any other option.

However, I would move this logic into a resolver, so create a new resolver & a query that uses this resolver. Then you can use inside that resolver the existing `SubgraphUserMotionTokenEvents` query. For example in `user` resolver file there is 
 a `getUserStakedBalance` function where we do similar processing (we just get the total user balance instead of the array of unclaimed motions).

Reply:
@chinins thank you, I have refactored to use a resolver. I ended up putting it in the `stakes.ts` resolver, as the user one was already quite big and I felt it made sense to have it in either.